app/src/main/java/com/vero/tasky/agenda/data/repository/EventRepositoryImpl.kt

```diff
@@ -0,0 +1,162 @@
+package com.vero.tasky.agenda.data.repository
+
+import com.vero.tasky.agenda.data.local.dao.EventDao
+import com.vero.tasky.agenda.data.local.dao.ModifiedAgendaItemDao
+import com.vero.tasky.agenda.data.local.entities.ModifiedAgendaItemEntity
+import com.vero.tasky.agenda.data.mappers.*
+import com.vero.tasky.agenda.data.remote.network.api.EventApi
+import com.vero.tasky.agenda.data.remote.network.dto.toAttendeeBaseInfo
+import com.vero.tasky.agenda.data.util.multipart.MultipartParser
+import com.vero.tasky.agenda.domain.model.*
+import com.vero.tasky.agenda.domain.repository.EventRepository
+import com.vero.tasky.core.data.remote.safeSuspendCall
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+class EventRepositoryImpl(
+    private val api: EventApi,
+    private val eventDao: EventDao,
+    private val modifiedAgendaItemDao: ModifiedAgendaItemDao,
+    private val multipartParser: MultipartParser
+) : EventRepository {
+
+    override suspend fun createEvent(
+        event: AgendaItem.Event,
+    ): Result<AgendaItemUploadResult> {
+        val localPhotos = event.photos.filterIsInstance<AgendaPhoto.LocalPhoto>()
+
+        saveEvent(event = event, localPhotos = localPhotos)
+
+        var skippedPhoto = 0
+
+        safeSuspendCall {
+            withContext(Dispatchers.IO) {
```

Comment:
It's not the responsibility of the caller of the suspend function to switch to the right context, but rather of the suspend function itself.
So in this case, it's the job of Retrofit to switch, since it executes blocking calls that should be run on the IO dispatcher. If you write your own suspend function which executes blocking non-suspending calls, you should switch the dispatcher there, so the function can be safely used from any dispatcher. This concept is called main-safety (safe to be called from the main thread)